fast64_internal/oot/oot_utility.py

```diff
@@ -874,6 +874,27 @@ def _eval(node):
     return f"0x{_eval(node.body):X}"
 
 
+def getShiftFromMask(mask: int):
+    """Returns the shift value from the mask"""
+
+    # get the shift by subtracting the length of the mask
+    # converted in binary on 16 bits (since the mask can be on 16 bits) with
+    # that length but with the rightmost zeros stripped
+    return int(f"{len(f'{mask:016b}') - len(f'{mask:016b}'.rstrip('0'))}", base=10)
+
+
+def getFormattedParams(mask: int, value: str, isBool: bool):
+    """Returns the parameter with the correct format"""
+    shift = getShiftFromMask(mask) if mask is not None else 0
+
+    if not int(getEvalParams(value), base=16):
```

Comment:
not sure what this means, you want to check for 0 ? `== 0` would be clearer

also `getEvalParams` should be modified to return an int instead of a string

or do something like:
```py
def getEvalParamsInt(...):
    ... current getEvalParams code minus stringify
def getEvalParams(...):
    i = getEvalParamsInt(...)
    return f"{i:X}"
```

Reply:
shift = shift_from_mask if there's a mask else if there's no mask then there's no shift